runtime/lua/vim/lsp/util.lua

```diff
@@ -660,8 +660,19 @@ do
       local start = diagnostic.range.start
       local finish = diagnostic.range["end"]
 
+      local hl = underline_highlight_name
+      if diagnostic.severity == protocol.DiagnosticSeverity.Error then
+          hl = underline_highlight_name..'Error'
```

Comment:
You will also need to add defaults like at 550. Probably use default link:

    hi default link LspDiagnosticsUnderlineError LspDiagnosticsUnderline

Reply:
thanks! fixed in new commit